src/api/app/lib/backend/proxy_request.rb

```diff
@@ -0,0 +1,136 @@
+      if @request.form_data?
+        # it's uncommon, but possible that we have both
+        query_string += '&' if query_string.present?
+        query_string += @request.raw_post
+      end
+      query_string = '?' + query_string if query_string.present?
+      path + query_string
+    end
+
+    def use_web_server_forward
+      # apache & mod_xforward case
+      if CONFIG['use_xforward'] && CONFIG['use_xforward'] != 'false'
+        logger.debug "[backend] VOLLEY(mod_xforward): #{@path}"
+        headers['X-Forward'] = "http://#{CONFIG['source_host']}:#{CONFIG['source_port']}#{@path}"
+        headers['Cache-Control'] = 'no-transform' # avoid compression
+        head(200)
+        @skip_validation = true
+        return true
+      end
+
+      # lighttpd 1.5 case
+      if CONFIG['x_rewrite_host']
+        logger.debug "[backend] VOLLEY(lighttpd): #{@path}"
+        headers['X-Rewrite-URI'] = @path
+        headers['X-Rewrite-Host'] = CONFIG['x_rewrite_host']
+        headers['Cache-Control'] = 'no-transform' # avoid compression
+        head(200)
+        @skip_validation = true
+        return true
+      end
+
+      # nginx case
+      if CONFIG['use_nginx_redirect']
+        logger.debug "[backend] VOLLEY(nginx): #{@path}"
+        headers['X-Accel-Redirect'] = "#{CONFIG['use_nginx_redirect']}/http/#{CONFIG['source_host']}:#{CONFIG['source_port']}#{@path}"
+        headers['Cache-Control'] = 'no-transform' # avoid compression
+        head(200)
+        @skip_validation = true
+        return true
+      end
+
+      false
+    end
+
+    # FIXME: This shares a lot of logic with Backend::File, should be merged.
+    def proxy_from_backend
+      logger.debug "[backend] VOLLEY: #{@path}"
+      Backend::Test.start
+      backend_http = Net::HTTP.new(CONFIG['source_host'], CONFIG['source_port'])
+      backend_http.read_timeout = 1000
+
+      # we have to be careful with object life cycle. the actual data is
+      # deleted once the tempfile is garbage collected, but isn't kept alive
+      # as the send_file function only references the path to it. So we keep it
+      # for ourselves. And once the controller is garbage collected, it should
+      # be fine to unlink the data
+      volleyfile = Tempfile.new('volley', Dir.tmpdir, encoding: 'ascii-8bit')
+      opts = { url_based_filename: true }
+
+      backend_http.request_get(path) do |response|
+        opts[:status] = response.code
+        opts[:type] = response['Content-Type']
+        response.read_body do |segment|
+          volleyfile.write(segment)
+        end
+      end
+      opts[:length] = volleyfile.length
+      opts[:disposition] = 'inline' if ['text/plain', 'text/xml'].include?(opts[:type])
+      # streaming makes it very hard for test cases to verify output
+      opts[:stream] = false if Rails.env.test?
+      send_file(volleyfile.path, opts)
+      # close the file so it's not staying in the file system
+      volleyfile.close
```

Comment:
it will just be deleted, if the `unlink` parameter is set to true https://ruby-doc.org/stdlib-2.5.3/libdoc/tempfile/rdoc/Tempfile.html#method-i-close otherwise the method `cache_request_body_to_file` wouldn't work at all.

Reply:
I try to limit the refactoring to clarification here.

Reply:
but the comment is misleading (aka wrong)

Reply:
Lots of things in here are wrong and misleading. I don't want to deal with those things now. In this PR I want to isolate the code.